table.go

```diff
@@ -366,6 +366,9 @@ func (t *Table) Iterator(
 				distinctColumns,
 			)
 			if err != nil {
+				if err == pqarrow.ErrColumNotFound {
```

Comment:
There's an edge case that I think we're forgetting about here, `column_that_doesnt_exist=""` should let us iterate over all of those that don't have the column.

Reply:
Yes, Javier and I were discussing that one too. Will need to write a test to see how this gets handled.